### DISCUSSION TITLE
77. 조인 연산이 많을 경우 성능이 저하되는 이유를 설명해 주세요.

### DISCUSSION THREAD

**Discussion by cutehumanS2** [jewelry-box/cs-interview#79] [General]
조인 연산이 많을 경우 성능이 저하되는 이유를 설명해 주세요.

**Comment by linirini** [comment]
- 조인은 여러 테이블의 데이터를 연결하여 결과를 생성하는 작업으로 그 과정에서 각 테이블의 데이터 간 관계를 확인하고, 조건에 맞는 데이터를 비교 및 필터링하는 작업이 필요하다.
- 조인이 많아질수록 카디널리티가 높아지므로, 데이터 비교 작업이 기하급수적으로 증가하여 성능에 부정적인 영향을 미친다.
- 조인 과정에서 중간 결과를 생성하기 위해 메모리를 사용한다. 중간 결과가 크면 메모리 사용량이 늘어나고 디스크 I/O가 발생할 수 있다.
- 조인에 사용되는 키에 인덱스가 없으면 데이터베이스가 조인 조건을 만족하는 행을 찾기 위해 테이블 전체를 스캔해야한다.
